renderer/components/Calendar/Calendar.tsx

```diff
@@ -139,13 +152,96 @@ export function Calendar({
           isValid: activities.every((report: ReportActivity) => report.isValid === true),
         };
       });
-      setMonthWorkHoursReports(monthWorkHours);
+
+      setFormattedQuarterReports(fromattedReports);
     } catch (err) {
       console.log("Error details ", err)
       setRenderError({errorTitle:"Calendar error", errorMessage:"An error occurred when validating reports for the last month. "})
+    }
+  }, [parsedQuarterReports, daysOff]);
 
-}
-  }, [monthReportsFromServer]);
+  const loadHolidaysAndVacations = async () => {
+    try {
+      const timetrackerUserInfo = JSON.parse(
+        localStorage.getItem("timetracker-user")
+      );
+      const plannerToken = timetrackerUserInfo?.accessToken;
+      const userPromises = [];
+
+      const holidaysPromise = global.ipcRenderer.invoke(
+        "timetracker:get-holidays",
+        plannerToken
+      );
+
+      const userEmail = timetrackerUserInfo?.email;
+      const vacationsPromise = global.ipcRenderer.invoke(
+        "timetracker:get-vacations",
+        plannerToken,
+        userEmail
+      );
+
+      userPromises.push(...[holidaysPromise, vacationsPromise]);
+
+      const userFetchedData = await Promise.all(userPromises);
+
+      if (userFetchedData.includes("invalid_token")) {
+        const refreshToken = timetrackerUserInfo?.refreshToken;
+
+        const refreshedPlannerCreds = await global.ipcRenderer.invoke(
+          "timetracker:refresh-planner-token",
+          refreshToken
+        );
+
+        const refreshedUserInfo = {
+          ...timetrackerUserInfo,
+          accessToken: refreshedPlannerCreds?.access_token,
+        };
+
+        localStorage.setItem(
+          "timetracker-user",
+          JSON.stringify(refreshedUserInfo)
+        );
+        loadHolidaysAndVacations();
+        return;
+      }
+
+      const holidays = userFetchedData[0];
+      const vacationsAndSickdays = userFetchedData[1]?.periods;
+      const userDaysOff: DayOff[] = [];
+
+      holidays.forEach((item) => {
+        userDaysOff.push({
+          date: new Date(item?.dateFrom),
+          duration: item?.quantity,
+          description: item?.description,
+        });
+      });
+
+      vacationsAndSickdays.forEach((item) => {
+        userDaysOff.push({
+          date: new Date(item?.dateFrom),
+          duration: item?.quantity,
+          description: item?.description,
+        });
+      });
```

Comment:
i would move it to another function
i would return data from `loadHolidaysAndVacations` and moved this function to utils, file like timetracker.ts and import it here
what will you say?